File: backend/astronomy_service.py
from fastapi import FastAPI, HTTPException
from pydantic import BaseModel
from base_service import create_service
from datetime import datetime
import random

app, run = create_service("Astronomy", 5003)

class AstronomyRequest(BaseModel):
    date_of_birth: str
    time_of_birth: str = None
    city_of_birth: str = None

@app.post("/astronomy")
async def get_astronomy(request: AstronomyRequest):
    try:
        # Convertir la fecha de nacimiento a objeto datetime
        birth_date = datetime.strptime(request.date_of_birth, "%Y-%m-%d")
        
        # Determinar la fase lunar basada en el día del mes
        day = birth_date.day
        if day < 8:
            moon_phase = "Luna Nueva"
        elif day < 15:
            moon_phase = "Cuarto Creciente"
        elif day < 22:
            moon_phase = "Luna Llena"
        else:
            moon_phase = "Cuarto Menguante"
        
        # Generar posiciones planetarias simuladas
        planets = ["Sol", "Luna", "Mercurio", "Venus", "Marte", "Júpiter", "Saturno", "Urano", "Neptuno"]
        planetary_positions = []
        
        for planet in planets:
            # Usar el nombre del planeta y la fecha como semilla para generar posiciones "deterministas"
            seed = sum(ord(c) for c in planet)
            date_seed = birth_date.toordinal()
            combined_seed = (seed + date_seed) % 1000
            
            # Generar valores basados en la semilla
            ra = f"{combined_seed % 24}h {combined_seed % 60}m"
            dec = f"{combined_seed % 90}° {combined_seed % 60}'"
            
            planetary_positions.append({
                "planet": planet,
                "ra": ra,
                "dec": dec
            })
        
        return {
            "moon_phase": moon_phase,
            "planetary_positions": planetary_positions
        }
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

if __name__ == "__main__":
    run()

File: lib/services/astronomy-service.ts
// Servicio de astronomía integrado en Next.js

interface PlanetaryPosition {
  planet: string
  ra: string
  dec: string
}

interface AstronomyResponse {
  moon_phase: string
  planetary_positions: PlanetaryPosition[]
}

export function getAstronomicalData(dateOfBirth: string): AstronomyResponse {
  // Simulación de fase lunar basada en el día del mes
  const day = Number.parseInt(dateOfBirth.split("-")[2])
  let moonPhase = "Luna Nueva"

  if (day < 8) {
    moonPhase = "Luna Nueva"
  } else if (day < 15) {
    moonPhase = "Cuarto Creciente"
  } else if (day < 22) {
    moonPhase = "Luna Llena"
  } else {
    moonPhase = "Cuarto Menguante"
  }

  // Simulación de posiciones planetarias
  const planets = ["Sol", "Luna", "Mercurio", "Venus", "Marte", "Júpiter", "Saturno", "Urano", "Neptuno"]
  const planetaryPositions = planets.map((planet) => {
    // Usar el nombre del planeta como semilla para generar posiciones "deterministas"
    const seed = planet.split("").reduce((acc, char) => acc + char.charCodeAt(0), 0)
    const dateValue = new Date(dateOfBirth).getTime()
    const combinedSeed = seed + dateValue

    // Generar valores basados en la semilla
    const ra = `${combinedSeed % 24}h ${combinedSeed % 60}m`
    const dec = `${combinedSeed % 90}° ${combinedSeed % 60}'`

    return {
      planet,
      ra,
      dec,
    }
  })

  return {
    moon_phase: moonPhase,
    planetary_positions: planetaryPositions,
  }
}

